cli/src/cli_util.rs

```diff
@@ -924,7 +932,35 @@ impl WorkspaceCommandHelper {
             // pointing to the new working-copy commit might not be exported.
             // In that situation, the ref would be conflicted anyway, so export
             // failure is okay.
-            self.snapshot_working_copy(ui)?;
+            match self.snapshot_working_copy(ui, update_stale)? {
+                SnapshotWorkingCopyResult::Success => {}
+                SnapshotWorkingCopyResult::UpdateStale => {
+                    match self.env.command.load_stale_working_copy_commit(ui)? {
+                        StaleWorkingCopy::Recovered((repo, _)) => {
+                            self.user_repo = ReadonlyUserRepo::new(repo);
+                        }
+                        StaleWorkingCopy::Snapshotted((repo, stale_commit)) => {
+                            let (locked_ws, new_commit) =
+                                self.unchecked_start_working_copy_mutation()?;
+                            let stats = update_stale_working_copy(
+                                locked_ws,
+                                repo.op_id().clone(),
+                                &stale_commit,
+                                &new_commit,
+                            )?;
+                            writeln!(
+                                ui.status(),
+                                "Automatically updated to fresh commit {}",
+                                short_commit_hash(stale_commit.id())
+                            )?;
+                            self.write_stale_commit_stats(ui, &new_commit, stats)?;
+
+                            self.user_repo = ReadonlyUserRepo::new(repo);
+                        }
```

Comment:
I think this can be simplified if auto-update-stale is handled by `CommandHelper::workspace_helper()`. Something like:
```rust
let maybe_stale = workspace_command.maybe_snapshot(ui)?;
let Some(stale_err) = maybe_stale else { return Ok(workspace_command); };
if !auto_update_stale {
   return Err(stale_err.into());
}
Self::load_stale_working_copy_and_recover_from_it(..) // this function will reload workspace and repo anyway
```